### PR TITLE
Lesson8 homework ab

### PR DESCRIPTION
**Pull request by somacruz10**

### CODE REVIEW

File: tests/calculatePasswordStrength.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { calculatePasswordStrength } from '../src/calculatePasswordStrength';
+import {LinkChecker} from "../src/url-checker"; // Укажите правильный путь к модулю
```
Comment (kapitonovalia) [inline tests/calculatePasswordStrength.test.ts:2]: import {LinkChecker} можно закомментить, если не используется

File: tests/calculatePasswordStrength.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        expect(calculatePasswordStrength('pass!123')).toBe('Moderate'); // Строчные, цифры, специальные символы
+        expect(calculatePasswordStrength('PAssWORD!')).toBe('Moderate'); // Прописные, строчные, цифры и специальные символы
+    });
+
+    test('should return "Strong" for passwords meeting 5 or more criteria', () => {
+        expect(calculatePasswordStrength('Pass1234!')).toBe('Strong'); // Прописные, строчные, цифры, спецсимволы, длина >= 8
+        expect(calculatePasswordStrength('Str0ngP@ssw0rd')).toBe('Strong'); // Все критерии, включая длину >= 12
+    });
+
+    test('should handle edge cases correctly', () => {
```
Comment (kapitonovalia) [inline tests/calculatePasswordStrength.test.ts:36]: Можно продолжить разделение проверок по блокам: разнести эти две проверки по блокам 'Very Weak' и 'Weak'.

File: tests/calculatePasswordStrength.test.ts
Comment (kapitonovalia) [inline tests/calculatePasswordStrength.test.ts:1]: Ты продумал тесты для разных уровней сложности паролей (от очень слабых до сильных), следуешь хорошей структуре для тестов с четкими названиями, что делает тесты легко читаемыми и понятными.

File: tests/calculatePasswordStrength.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        expect(calculatePasswordStrength('12345')).toBe('Very Weak');
+        expect(calculatePasswordStrength('password')).toBe('Very Weak'); // Только строчные буквы
+    });
+
+    test('should return "Weak" for passwords with basic length and one other criteria', () => {
+        expect(calculatePasswordStrength('123456789101112')).toBe('Weak'); // Только цифры
+        expect(calculatePasswordStrength('password1')).toBe('Weak'); // Буквы и цифра
+        expect(calculatePasswordStrength('PASSw1')).toBe('Weak'); // Большие буквы и цифра
+        expect(calculatePasswordStrength('PAS1!')).toBe('Weak'); // Большие буквы, цифра и символ
+        expect(calculatePasswordStrength('aaaaaaaaaaaa')).toBe('Weak'); // Только строчные буквы
```
Comment (kapitonovalia) [inline tests/calculatePasswordStrength.test.ts:22]: Для лучшей читаемости можно поднять проверку "Только строчные буквы" до/после проверки "Только цифры". т е разделить тесты (Проверки паролей длиной менее 8 символов, но соответствующие 3 критериям и проверки паролей с длиной больше 8 символов и удовлетворяющие 2 критериям)
